Fix data_summary() bug and add test

Aryllen opened the pull request:
Fixes #288.

Changes proposed in this pull request:

- Fix the `data_summary()` bug where empty columns resulted in an error
- Add tests

Please confirm you've done the following (if applicable):

- Added tests for new functions or for new behavior in existing functions: yes
- If adding a new exported function, added it to the reference section of `_pkgdown.yml`: N/A
- If adding a new configuration option, documented it in `vignettes/customizing_dccvalidator.Rmd`: N/A

karawoo commented:
Would it work to use `NA` instead of `NULL` in these lines within `summarize_values()` instead? Or does that cause other problems?

https://github.com/Sage-Bionetworks/dccvalidator/blob/f4201a1f20ddcbee3d77b61778f1ea8635ebf31b/R/mod-file-summary.R#L198-L200

Aryllen commented:
I can try that. It's not always sending back `NULL` for all of these, though. If there's more than one empty value, it sees a list and doesn't return `NULL`. I should add a check for that.

Aryllen commented:
I changed it so `summarize_values()` is more stable and returns `NA` if given `NA` or `NULL`. It works for a single NULL/NA or for a list of NULLs/NAs. Updated/added tests.
